### PR TITLE
FIX: Modify stacking logic in ComputationalRoutine

### LINKED ISSUE #130: inducer bug - spy.selectdata-
Thank you for taking the time to report an issue! We know that writing detailed error reports takes effort, and we appreciate you taking the time.

**Describe the bug**
image

image

image

**To Reproduce**
1. Load/Import dataset 
2. select only certain amount of channel with the fuction spy.selectdata(data, channel = [x,y,z])
3. Error message/erroneous result. 

**Thank you again for your time!**

### PR DESCRIPTION
**Pull request by pantaray**
- new mandatory positional argument in `initialize` of `ComputationalRoutine`:
  `out_dimord` provides the output object's dimensional labels. This is used
  to determine the correct stacking dimension for trials
- do not default to the first dimension as stacking dimension: instead use the
  new `stackingDim` variable (which is determined based on `out_dimord`, see above)
- closes #130

On branch comproutine_fix
 Changes to be committed:
	modified:   syncopy/connectivity/connectivity_analysis.py
	modified:   syncopy/datatype/methods/selectdata.py
	modified:   syncopy/shared/computational_routine.py
	modified:   syncopy/specest/freqanalysis.py
	modified:   syncopy/tests/spy_setup.py
	modified:   syncopy/tests/test_computationalroutine.py
	modified:   syncopy/tests/test_specest.py

Author Guidelines
-----------------
- [X] Is the change set **< 400 lines**?
- [X] Was the code checked for memory leaks/performance bottlenecks?
- [X] Is the code running locally **and** on the ESI cluster?
- [X] Is the code running on all supported platforms?

Reviewer Checklist
------------------
- [ ] Are testing routines present?
- [ ] Do **parallel** loops have a set length and correct termination conditions?
- [ ] Do objects in the global package namespace perform proper parsing of their input? 
- [ ] Do code-blocks provide novel functionality, i.e., no re-factoring using builtin/external packages possible?
- [ ] Code layout
  - [ ] Is the code PEP8 compliant?
  - [ ] Does the code adhere to agreed-upon naming conventions?
  - [ ] Are keywords clearly named and easy to understand?
  - [ ] No commented-out code?
- [ ] Are all docstrings complete and accurate?

### CODE REVIEW

File: syncopy/shared/computational_routine.py [hunk 1]
```diff
@@ -291,17 +293,34 @@ def initialize(self, data, chan_per_worker=None, keeptrials=True):
             dtp_list.append(dtype)
             trials.append(trial)
 
+        # Determine trial stacking dimension and compute aggregate shape of output
+        if "time" in out_dimord:        # AnalogData + SpectralData + CrossSpectralData
```
Comment (tensionhead) [inline syncopy/shared/computational_routine.py:297]: Might be that we have to come back here and add more variables?! Would it be maybe better to define a 'stackingDim' in the respective DataClass? It anyways knows it's own dimord of course, and hence can also inform the CR about it? It looks like the only use of the `out_dimord` is anyways exactly this no?

File: syncopy/tests/test_specest.py [hunk 2]
```diff
@@ -918,6 +918,7 @@ class TestWavelet():
                        "channels": range(0, int(nChannels / 2)),
                        "toilim": [-20, 60.8]}]
 
+    @skip_low_mem
```
Comment (tensionhead) [inline syncopy/tests/test_specest.py:921]: Uh this magic is new to me :)

File: syncopy/shared/computational_routine.py (hunk 1 above)
Comment (pantaray) [reply syncopy/shared/computational_routine.py:297]: Jupp, absolutley right. Godd point. We could add a `stackingDim` attribute to each of our classes - that might make things more transparent too. I would actually prefer this approach. Do you want me to go ahead and implement it?

File: syncopy/tests/test_specest.py (hunk 2 above)
Comment (pantaray) [reply syncopy/tests/test_specest.py:921]: :D Let's see if that helps w/testing pipeline stability on low-spec systems. Fingers crossed!

File: syncopy/shared/computational_routine.py (hunk 1 above)
Comment (tensionhead) [reply syncopy/shared/computational_routine.py:297]: Yes please go ahead :smiley: !

Comment (pantaray) [reply syncopy/shared/computational_routine.py]: Done, please re-review

File: syncopy/shared/computational_routine.py [hunk 3]
```diff
@@ -294,17 +294,12 @@ def initialize(self, data, out_dimord, chan_per_worker=None, keeptrials=True):
             trials.append(trial)
 
         # Determine trial stacking dimension and compute aggregate shape of output
-        if "time" in out_dimord:        # AnalogData + SpectralData + CrossSpectralData
```
Comment (tensionhead) [inline syncopy/shared/computational_routine.py]: Perfect, also directly looks better :)

**Review by tensionhead — APPROVED** [review]

**Review by tensionhead — APPROVED** [review]
That's it!

**Pull request merged**